record/record.go

```diff
@@ -198,9 +206,27 @@ func (r *Reader) Next() (io.Reader, error) {
 		return nil, r.err
 	}
 	r.started = true
+	r.inRecovery = false
 	return singleReader{r, r.seq}, nil
 }
 
+// Recover clears any errors read so far, so that calling Next will start
+// reading from the next good 32KiB block. If there are no such blocks, Next
+// will return io.EOF.
+//
+// Recover also marks the current reader, the one most recently returned by
+// Next, as stale.
+func (r *Reader) Recover() {
+	// After calling r.Recover, we're effectively starting fresh, so we need to
+	// set r.inRecovery to true so r.Next and r.nextChunk proceed with the next block.
+	// We also set r.{i,j,n} = 0 and r.err = nil so we don't reparse the old block's data.
+	r.i, r.j, r.n = 0, 0, 0
+	r.err = nil
+	r.inRecovery = true
```

Comment:
You might not need the inRecovery field. Can you just say "r.started = false" here??